week-2-wide-perspective/w02e05-simple-server-side/src/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,26 @@
+import { NextFunction, Request, Response } from "express";
+import jwt from "jsonwebtoken";
+
+export async function authMiddleware(req: Request, res: Response, next: NextFunction) {
+  const authHeader = req.headers.authorization;
+
+  if (!authHeader) {
+    res.status(401).send("Unauthorized");
+  } else {
+    const [_, token] = authHeader.split(" ");
+    const { JWT_SECRET_TOKEN = "supersecret" } = process.env;
+
+    const decodedToken = jwt.verify(token, JWT_SECRET_TOKEN);
+    const { id, email } = decodedToken as { id: number; email: string };
+
+    if (!decodedToken) {
+      res.status(401).send("Unauthorized");
```

Comment:
Twoje `.send` to problem.
Ponieważ zmieniasz sposób komunikacji z `frontem`

`95%` Twoich wiadomości wysyłanych przez serwer to `JSON` i to ok, bo robiłeś `API`
Ale tutaj - wysyłasz `plain-text` !

Dodatkowo jeśli wystąpi jakiś błąd, to wbudowany "error catcher" też wyśle `plain-text` - chyba że go "napsiszesz" swoim (szczegóły wyżej)